avro/src/main/java/com/fasterxml/jackson/dataformat/avro/schema/AvroSchemaHelper.java

```diff
@@ -332,9 +341,44 @@ public static String getFullName(Schema schema) {
                 return namespace + name;
             }
             StringBuilder sb = new StringBuilder(1 + namespace.length() + name.length());
-            return sb.append(namespace).append('.').append(name).toString();
-        default:
+            // Check if this is a nested class
+            String nestedClassName = sb.append(namespace).append('$').append(name).toString();
+            try {
+                Class.forName(nestedClassName);
```

Comment:
@baharclerode assuming most common case is NOT inner class, would it make sense to try loading "regular" class first, inverting logic? Only asking since there's quite a bit of overhead for exception handling and possibly other corner cases for trying to load non-existing class.

Reply:
I don't think so; Either way, we're always doing at least 1 `Class.forName()`; Either we check for the non-nested class, and assume if that fails that it's nested, or we always check for the nested class, and if that fails, assume that it's not-nested.

Reply:
Alas, looks like this lead to a major performance issue for some usage (see #219 for attempts to work around the issue). I assume that case of finding class is relatively cheap, but that failure is enormously expensive (as that is the "rare" case being optimized against), depending on complexity of classpath for application.